client_test.go

```diff
@@ -1,38 +1,91 @@
-package apns_test
+package apns
 
 import (
-	"bytes"
-	"encoding/binary"
+	"errors"
 	"io/ioutil"
 	"os"
-	"time"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/timehop/apns"
+	"github.com/timehop/tcptest"
 )
 
+type mockSession struct {
```

Comment:
is mockSession something that users of this library would find useful when testing?